editor/editor_inspector.cpp

```diff
@@ -3064,46 +3066,52 @@ void EditorInspector::update_tree() {
 				path = vformat(TTR("Element %s"), array_index);
 			}
 		} else {
-			// Check if we exit or not a subgroup. If there is a prefix, remove it from the property label string.
-			if (!subgroup.is_empty() && !subgroup_base.is_empty()) {
-				if (path.begins_with(subgroup_base)) {
-					path = path.trim_prefix(subgroup_base);
-				} else if (subgroup_base.begins_with(path)) {
-					// Keep it, this is used pretty often.
+			// Detect whether the property is in the active group/subgroup
+			// based on its name. If it is, transform its name into a path
+			// that reflects its membership.
+			//
+			// e.g. if we're building a group with display name 'Process'
+			// and prefix 'process_', then the path 'process_mode' will get
+			// rewritten to 'Process/mode'.
+
+			// Check if we're exiting a subgroup.
+			if (!active_subgroup_display_name.is_empty() && !active_subgroup_prefix.is_empty()) {
+				if (path.begins_with(active_subgroup_prefix)) {
+					path = path.trim_prefix(active_subgroup_prefix);
+				} else if (active_subgroup_prefix.begins_with(path)) {
+					// We're still in the subgroup.
 				} else {
-					subgroup = ""; // The prefix changed, we are no longer in the subgroup.
+					active_subgroup_display_name = "";
 				}
 			}
 
-			// Check if we exit or not a group. If there is a prefix, remove it from the property label string.
-			if (!group.is_empty() && !group_base.is_empty() && subgroup.is_empty()) {
-				if (path.begins_with(group_base)) {
-					path = path.trim_prefix(group_base);
-				} else if (group_base.begins_with(path)) {
-					// Keep it, this is used pretty often.
+			// Check if we're exiting a group.
+			if (!active_group_display_name.is_empty() && !active_group_prefix.is_empty() && active_subgroup_display_name.is_empty()) {
+				if (path.begins_with(active_group_prefix)) {
+					path = path.trim_prefix(active_group_prefix);
+				} else if (active_group_prefix.begins_with(path)) {
+					// We're still in the group.
 				} else {
-					group = ""; // The prefix changed, we are no longer in the group.
-					subgroup = "";
+					active_group_display_name = "";
+					active_subgroup_display_name = "";
 				}
 			}
 
-			// Add the group and subgroup to the path.
-			if (!subgroup.is_empty()) {
-				path = subgroup + "/" + path;
+			if (!active_subgroup_display_name.is_empty()) {
+				path = active_subgroup_display_name + "/" + path;
 			}
-			if (!group.is_empty()) {
-				path = group + "/" + path;
+			if (!active_group_display_name.is_empty()) {
+				path = active_group_display_name + "/" + path;
 			}
 		}
 
-		// Get the property label's string.
-		String name_override = (path.contains("/")) ? path.substr(path.rfind("/") + 1) : path;
```

Comment:
This variable is not the raw property name.

It's "overriding" because it stores the name without the optional feature tag part, and the inspector later uses this to reassemble a pseudo property name/path without the feature tag.

Reply:
reverted, thank you!